pkg/client/typed/dynamic/client.go

```diff
@@ -83,6 +83,13 @@ func (c *Client) Resource(resource *unversioned.APIResource, namespace string) *
 	}
 }
 
+func (c *Client) Subresource(resource *unversioned.APIResource, subresource string, namespace string) *SubresourceClient {
```

Comment:
I haven't followed conventions in the new package, but this reads ugly to my eyes.  Why don't we have `func (c *ResourceClient) Subresource(subresource string) *SubresourceClient` so that I can do `.Resource(resource, namespace).Subresource("scale")`?  That seems to contain the information more cleanly rather than plumbing everything across every layer.